engine/src/test/java/io/zeebe/engine/processing/streamprocessor/ReplayStateTest.java

```diff
@@ -258,6 +259,156 @@
                       .withElementType(BpmnElementType.END_EVENT)
                       .withElementId("end")
                       .getFirst();
+                }),
+        testCase("parallel multi-instance service task")
+            .withProcess(
+                Bpmn.createExecutableProcess(PROCESS_ID)
+                    .startEvent()
+                    .serviceTask(
+                        "task",
+                        t ->
+                            t.zeebeJobType("type")
+                                .multiInstance(
+                                    m ->
+                                        m.parallel()
+                                            .zeebeInputElement("item")
+                                            .zeebeInputCollectionExpression("items")
+                                            .zeebeOutputElementExpression("result")
+                                            .zeebeOutputCollection("results")))
+                    .endEvent()
+                    .done())
+            .withExecution(
+                engine -> {
+                  final long piKey =
+                      engine
+                          .processInstance()
+                          .ofBpmnProcessId(PROCESS_ID)
+                          .withVariable("items", List.of(1, 2, 3))
+                          .create();
+                  Awaitility.await("until there are 3 jobs ready to be activated")
+                      .pollInSameThread()
+                      .until(
+                          () ->
+                              RecordingExporter.jobRecords(JobIntent.CREATED).limit(3).count()
+                                  >= 3);
+                  final JobBatchRecordValue jobs =
+                      engine.jobs().withMaxJobsToActivate(3).withType("type").activate().getValue();
+                  jobs.getJobKeys()
+                      .forEach(
+                          key -> engine.job().withKey(key).withVariable("result", 0).complete());
+
+                  return RecordingExporter.processInstanceRecords(
+                          ProcessInstanceIntent.ELEMENT_COMPLETED)
+                      .withProcessInstanceKey(piKey)
+                      .withElementType(BpmnElementType.PROCESS)
+                      .getFirst();
+                }),
+        testCase("sequential multi-instance service task")
```

Comment:
This test lies :cake: 

In the process, the multi-instance should be sequential, not parallel.

Reply:
The dangers of copy pasting :see_no_evil: